kong/api/routes/upstreams.lua

```diff
@@ -160,4 +103,25 @@ return {
     end
   },
 
+  ["/upstreams/:name_or_id/targets/:target"] = {
+    before = function(self, dao_factory, helpers)
+      crud.find_upstream_by_name_or_id(self, dao_factory, helpers)
+      self.params.upstream_id = self.upstream.id
+    end,
+
+    DELETE = function(self, dao_factory)
+      balancer.clean_history(self.params.upstream_id, dao_factory)
+
+      -- this is just a wrapper around POSTing a new target with weight=0
+      local target  = self.params
+      target.weight = 0
+
+      local data, err = dao_factory.targets:insert(target)
+      if err then
+        return app_helpers.yield_error(err)
+      end
+
+      return responses.send_HTTP_NO_CONTENT()
```

Comment:
I think if we were to return 201 here, we might as well not bother with this method at all ;)

tbh, i dont even think this is a valid PR. I don't think adding this endpoint makes it any easier for consumers to work with the upstream/target interface. But that's just my $0.02 ;)